### ISSUE TITLE
Intellisense tooltip with filter category Visual Studio like

### ISSUE THREAD

**Issue by Jonatthu** [Microsoft/vscode#50291]
![](https://www.visualstudio.com/wp-content/uploads/2018/05/assisted-intellisense@2x-800x436.jpg)

It would be awesome if in the moment to bring the intellisense of properties or classes on Typescript and C# or other language we can filter this like the little icons on the bottom of the tooltip as visual studio,
or control + shift + p + @: but for the intellisense.

**Comment by mjbvz** [comment]
Likely duplicate of #18639, although this issue is clearer

**Comment by Jonatthu** [comment]
@svipben I like those ideas, but I think this is a must to have, usually I get all mixed including snippets, I can order by Name and the current features but it is not the same enjoying experience as Visual Studio and gets hard to do quick choices.

**Comment by jul1u5** [comment]
I would like to try to implement this feature. I've found the [directory](https://github.com/Microsoft/vscode/blob/master/src/vs/editor/contrib/suggest) in which I think I should create this filtering behaviour. I don't know, though, where in particular this feature should reside. It would be best if someone would give a quick high-level overview of `suggest` folder.

**Comment by jrieken** [comment]
The filtering is pretty easy, everything happens in the [`_createCachedState`](https://github.com/Microsoft/vscode/blob/a46d73ebed315f6eef4278d62734ecd4fce38f73/src/vs/editor/contrib/suggest/completionModel.ts#L149)-method. Wiring this into the UI is the bigger chunk, @ramya-rao-a and @joaomoreno might be able to help. 

But before all of this we should discuss if we actually want this feature in VS Code. So far this feature request hasn't picked up many thumbs-up...

**Comment by jul1u5** [comment]
There is already a [question](https://stackoverflow.com/questions/44011708/mimic-visual-studio-intellisense-filtering-behavior-in-visual-studio-code) on stack overflow about this. It might clutter the suggestion box, though. Maybe we could make this an opt-in feature (like breadcrumbs).

**Comment by Jonatthu** [comment]
@jrieken I do not think we need a massive number of thumbs up to know what is going to be really useful.

**Comment by DeltaRazero** [comment]
Bump, would really like to see this implemented in the future.

To further expand the tooltip filter idea:
I'm using mingw-w64 for C++ dev and whenever I want to #include something I get a very big list, but there are also directories in my include directory and it would be nice to just to view the directories instead of the huge list with all header files included.

**Comment by theo-bittencourt** [comment]
Without that feature, the Intellisense autocomplete is *unuseful* when you have a lot of suggestions.

**Comment by mitsukuri** [comment]
This has to be bumped at least once in two years.

**Comment by Braveo** [comment]
Bumping this up for C# development since it'd be nice if we can filter for event handlers.

**Comment by TheGrowingPlant443** [comment]
I add in that it would be nice in Python as well. I get methods before variabels and it not fun when you have 50 methods and 3 variables on a object.

**Comment by aderchox** [comment]
I was going to open a new issue for this, but then I found this one here.
Thanks @Jonatthu. This is really an "annoying" missing feature of VSCode.
I only want to add something else too.
Having a keybinding for filtering the type of the suggestions (e.g., filter only the methods) would be even better (compared to the icons shown at the bottom of the suggestion box, in case of the Visual Studio). Of course, both of these have no conflicts with each other. The reason I'm asking for a keybinding is you won't need to leave the keyboard to filter the suggestions.
The keybinding must work only when the suggestions box is active.

**Comment by JaneySprings** [comment]
It would be great to have such a feature for c# dev

**Comment by ThaDaVos** [comment]
Whoaw years later and this isn't done yet - would be great if this was done the same way as in Visual Studio - just the little filter icons at the bottom would help a lot - especially in C#

**Comment by jasenf** [comment]
Agree.  This is a hugely used feature in VS2022.   Trying to make my permanent migration to VSCode and miss this feature a ton.

**Comment by Niproblema** [comment]
Would be great

**Comment by Tangent-90** [comment]
any updates?

**Comment by svenwanzenried** [comment]
Are there any plans on implementing this? @jrieken What is your current take on this? Was there a discussion anywhere in the last 6 years if this would be a useful feature?
I myself would consider this feature a huge benefit for the editor overall.

**Comment by adzierzanowski** [comment]
I made a simple extension for that. It depends on changing the workspace settings so it's not the most efficient thing in the world but it's usable.

https://marketplace.visualstudio.com/items?itemName=doublefloat.intellisensefilter

**Comment by DonutWrecker** [comment]
It's mind-boggling that this feature is still not implemented.

**Comment by sacha-renault** [comment]
Still needed, years later.
